src/meta/src/manager/catalog/fragment.rs

```diff
@@ -165,7 +165,10 @@ where
 
     async fn notify_fragment_mapping(&self, table_fragment: &TableFragments, operation: Operation) {
         for fragment in table_fragment.fragments.values() {
-            if !fragment.state_table_ids.is_empty() {
+            if !fragment.state_table_ids.is_empty()
+                // DML fragment mapping is used by frontend to decide where to execute.
+                || fragment.fragment_type_mask & FragmentTypeFlag::Dml as u32 != 0
```

Comment:
I'm considering whether we can always notify mappings from all fragments here. A fragment will rarely have no stateful or DML operators. 😂